### PR TITLE
support yarn pnp for nodeAddonIncludes

### PR DESCRIPTION
**Pull request by Mesteery**
This solves problems with Yarn PnP and node addon includes.

*✅ Works correctly.*

### CODE REVIEW

**Comment by sean-mcmanus** [comment]
Thanks for the contribution. We'll try to review this later this week.

**Review by sean-mcmanus — CHANGES_REQUESTED** [review]
The stdout I get from yarn is
```
'yarn node v1.19.1\n"<workspaceFolder>\node_modules\\node-addon-api"\nDone in 0.51s.'
```
which isn't processed correctly because the code just expects the middle line. Is that supposed to work correctly and/or how are you getting it to work?

File: Extension/src/LanguageServer/configurations.ts [hunk 1]
```diff
@@ -406,15 +406,23 @@ export class CppProperties {
...
-                    "node-addon-api": `"${pathToNode}" --no-warnings -p "require('node-addon-api').include"`
-                };
+                const nodeAddonMap: [string, string][] = [
+                    ["node-addon-api", `"${pathToNode}" --no-warnings -p "require('node-addon-api').include"`],
+                    ["nan", `"${pathToNode}" --no-warnings -e "require('nan')"`]
+                ];
+                // Yarn (2) PnP support
+                const pathToYarn: string | null = which.sync("yarn", { nothrow: true });
+                if (pathToYarn && await util.checkDirectoryExists(path.join(rootPath, ".yarn/cache"))) {
+                    nodeAddonMap.push(
```
Comment (sean-mcmanus) [inline Extension/src/LanguageServer/configurations.ts:416]: Also, this folder/file doesn't exist for me.

**Comment by Mesteery** [comment]
This is normal, because it's for Yarn Berry.
https://yarnpkg.com/getting-started/install

**Comment by sean-mcmanus** [comment]
> This is normal, because it's for Yarn Berry.
> https://yarnpkg.com/getting-started/install

Ah, thanks. I'll look into that tomorrow.

**Review by sean-mcmanus — APPROVED** [review]

**Pull request merged**